docs/contributing/testing/database/index.md

```diff
@@ -0,0 +1,82 @@
+# Database Integration Testing
+
+Since Bitwarden has multiple database options, testing them all automatically is incredibly
+important so that we don't have to have a full QA process for each database type. This is where the
+`Infrastructure.IntegrationTest` project comes in, allowing the setup of tests similarly to how the
+databases are consumed in the applications through their common interfaces, generally an
+`I*Repository`. These tests are automatically executed on all supported databases in our testing
+pipeline.
+
+## Creating a new test
+
+To create a new database test just add the `[DatabaseTheory]` and `[DatabaseData]` attributes to
+test. Now, in the parameters of the test you are able to "inject" any repository layer services
+directly into the test. The test will run for every database that is
+[configured in the current environment](#configure-the-tests). Since you inject the interface of the
+service, some runs of the test will use the Dapper based repository implementation targeting
+Microsoft SQL Server and others will use the Entity Framework Core based implementations, which we
+use for MySql, Postgres, and SQLite.
```

Comment:
```suggestion
To create a new database test, add the `[DatabaseTheory]` and `[DatabaseData]` attributes to
test. Then, use the parameters of the test to inject any repository layer services
you need. The test will run for every database that is
[configured in the current environment](#configure-the-tests). Since you inject the interface of the
service, some runs of the test will use the Dapper-based repository implementation targeting
Microsoft SQL Server and others will use the Entity Framework Core based implementations (which we
use for MySql, Postgres, and SQLite).
```